activerecord/lib/active_record/associations/collection_association.rb

```diff
@@ -642,6 +644,14 @@ def first_nth_or_last(type, *args)
             set_inverse_instance record if record.is_a? ActiveRecord::Base
           end
         end
+
+        def associate_all_records(target)
+          if owner.new_record? && target.all? {|record| record.new_record? }
```

Comment:
That line is only run when all of the records are new ie both the owner and it's children or child. It's never run otherwise.